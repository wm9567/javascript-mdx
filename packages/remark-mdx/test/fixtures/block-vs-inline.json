{
  "type": "root",
  "children": [
    {
      "type": "heading",
      "depth": 1,
      "children": [
        {
          "type": "text",
          "value": "Block vs. inline",
          "position": {
            "start": {
              "line": 1,
              "column": 3,
              "offset": 2
            },
            "end": {
              "line": 1,
              "column": 19,
              "offset": 18
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 1,
          "column": 1,
          "offset": 0
        },
        "end": {
          "line": 1,
          "column": 19,
          "offset": 18
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "To make this example more readable, uppercase letters are used when JSX is\nparsed as a block, wheres lowercase letters are used for inline JSX.",
          "position": {
            "start": {
              "line": 3,
              "column": 1,
              "offset": 20
            },
            "end": {
              "line": 4,
              "column": 69,
              "offset": 163
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 3,
          "column": 1,
          "offset": 20
        },
        "end": {
          "line": 4,
          "column": 69,
          "offset": 163
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "This is inline, itâ€™s a paragraph, because ",
          "position": {
            "start": {
              "line": 6,
              "column": 1,
              "offset": 165
            },
            "end": {
              "line": 6,
              "column": 43,
              "offset": 207
            }
          }
        },
        {
          "type": "inlineCode",
          "value": "<x>",
          "position": {
            "start": {
              "line": 6,
              "column": 43,
              "offset": 207
            },
            "end": {
              "line": 6,
              "column": 48,
              "offset": 212
            }
          }
        },
        {
          "type": "text",
          "value": " is followed by ",
          "position": {
            "start": {
              "line": 6,
              "column": 48,
              "offset": 212
            },
            "end": {
              "line": 6,
              "column": 64,
              "offset": 228
            }
          }
        },
        {
          "type": "inlineCode",
          "value": "_",
          "position": {
            "start": {
              "line": 6,
              "column": 64,
              "offset": 228
            },
            "end": {
              "line": 6,
              "column": 67,
              "offset": 231
            }
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 6,
              "column": 67,
              "offset": 231
            },
            "end": {
              "line": 6,
              "column": 68,
              "offset": 232
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 6,
          "column": 1,
          "offset": 165
        },
        "end": {
          "line": 6,
          "column": 68,
          "offset": 232
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "mdxJsxTextElement",
          "name": "x",
          "attributes": [],
          "children": [
            {
              "type": "emphasis",
              "children": [
                {
                  "type": "text",
                  "value": "alpha",
                  "position": {
                    "start": {
                      "line": 8,
                      "column": 5,
                      "offset": 238
                    },
                    "end": {
                      "line": 8,
                      "column": 10,
                      "offset": 243
                    }
                  }
                },
                {
                  "type": "mdxJsxTextElement",
                  "name": "y",
                  "attributes": [],
                  "children": [],
                  "position": {
                    "start": {
                      "line": 8,
                      "column": 10,
                      "offset": 243
                    },
                    "end": {
                      "line": 8,
                      "column": 14,
                      "offset": 247
                    }
                  }
                }
              ],
              "position": {
                "start": {
                  "line": 8,
                  "column": 4,
                  "offset": 237
                },
                "end": {
                  "line": 8,
                  "column": 15,
                  "offset": 248
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 8,
              "column": 1,
              "offset": 234
            },
            "end": {
              "line": 8,
              "column": 19,
              "offset": 252
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 8,
          "column": 1,
          "offset": 234
        },
        "end": {
          "line": 8,
          "column": 19,
          "offset": 252
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "This is inline too, not a heading:",
          "position": {
            "start": {
              "line": 10,
              "column": 1,
              "offset": 254
            },
            "end": {
              "line": 10,
              "column": 35,
              "offset": 288
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 10,
          "column": 1,
          "offset": 254
        },
        "end": {
          "line": 10,
          "column": 35,
          "offset": 288
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "mdxJsxTextElement",
          "name": "x",
          "attributes": [],
          "children": [
            {
              "type": "text",
              "value": "# hea\nding",
              "position": {
                "start": {
                  "line": 12,
                  "column": 4,
                  "offset": 293
                },
                "end": {
                  "line": 13,
                  "column": 5,
                  "offset": 303
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 12,
              "column": 1,
              "offset": 290
            },
            "end": {
              "line": 13,
              "column": 9,
              "offset": 307
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 12,
          "column": 1,
          "offset": 290
        },
        "end": {
          "line": 13,
          "column": 9,
          "offset": 307
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "This is a block: there is nothing before or after the tags.",
          "position": {
            "start": {
              "line": 15,
              "column": 1,
              "offset": 309
            },
            "end": {
              "line": 15,
              "column": 60,
              "offset": 368
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 15,
          "column": 1,
          "offset": 309
        },
        "end": {
          "line": 15,
          "column": 60,
          "offset": 368
        }
      }
    },
    {
      "type": "mdxJsxFlowElement",
      "name": "X",
      "attributes": [],
      "children": [
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "One paragraph.",
              "position": {
                "start": {
                  "line": 18,
                  "column": 3,
                  "offset": 376
                },
                "end": {
                  "line": 18,
                  "column": 17,
                  "offset": 390
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 18,
              "column": 3,
              "offset": 376
            },
            "end": {
              "line": 18,
              "column": 17,
              "offset": 390
            }
          }
        },
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "Two paragraph.",
              "position": {
                "start": {
                  "line": 20,
                  "column": 3,
                  "offset": 394
                },
                "end": {
                  "line": 20,
                  "column": 17,
                  "offset": 408
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 20,
              "column": 3,
              "offset": 394
            },
            "end": {
              "line": 20,
              "column": 17,
              "offset": 408
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 17,
          "column": 1,
          "offset": 370
        },
        "end": {
          "line": 21,
          "column": 5,
          "offset": 413
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Blocks can be self-closing:",
          "position": {
            "start": {
              "line": 23,
              "column": 1,
              "offset": 415
            },
            "end": {
              "line": 23,
              "column": 28,
              "offset": 442
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 23,
          "column": 1,
          "offset": 415
        },
        "end": {
          "line": 23,
          "column": 28,
          "offset": 442
        }
      }
    },
    {
      "type": "mdxJsxFlowElement",
      "name": "X",
      "attributes": [],
      "children": [],
      "position": {
        "start": {
          "line": 25,
          "column": 1,
          "offset": 444
        },
        "end": {
          "line": 25,
          "column": 5,
          "offset": 448
        }
      }
    }
  ],
  "position": {
    "start": {
      "line": 1,
      "column": 1,
      "offset": 0
    },
    "end": {
      "line": 26,
      "column": 1,
      "offset": 449
    }
  }
}
