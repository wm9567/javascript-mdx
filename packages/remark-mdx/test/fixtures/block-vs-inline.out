# Block vs. inline

To make this example more readable, uppercase letters are used when JSX is
parsed as a block, wheres lowercase letters are used for inline JSX.

This is inline, itâ€™s a paragraph, because `<x>` is followed by `_`.

<x>*alpha<y/>*</x>

This is inline too, not a heading:

<x># hea
ding</x>

This is a block: there is nothing before or after the tags.

<X>
  One paragraph.

  Two paragraph.
</X>

Blocks can be self-closing:

<X/>
